### PR TITLE
fix: Add AllowDynamicProperties annotation in Service class

### PR DESCRIPTION
**Pull request by ajupazhamayil**
Resolves https://github.com/googleapis/google-api-php-client-services/issues/4106

### CODE REVIEW

File: src/Client.php [hunk 1]
```diff
@@ -49,6 +49,7 @@
  * The Google API Client
  * https://github.com/google/google-api-php-client
  */
+#[\AllowDynamicProperties]
```
Comment (bshaffer) [inline src/Client.php:52]: Why are we adding dynamic properties to this class? This class is not extended by anything, nor does it extend anything. So any dynamic properties it has should be defined.

Are you seeing dynamic property warnings anywhere for this class?

**Review by bshaffer — CHANGES_REQUESTED** [review]

File: src/Client.php (hunk 1 above)
Comment (ajupazhamayil) [reply src/Client.php:52]: You are right, removing this change.

**Review by bshaffer — COMMENTED** [review]
Just to be clear before we merge this - where is it that dynamic properties are being used for `Service` classes? Can you identify where it's happening (so we can also confirm that this is the right action to take)?

**Comment by ajupazhamayil** [comment]
> Just to be clear before we merge this - where is it that dynamic properties are being used for `Service` classes? Can you identify where it's happening (so we can also confirm that this is the right action to take)?

Hmm.. I was thinking about the `ShoppingContent` or similar classes extending the `Service` class. But, I just looked at them and the classes I should be concentrating are classes which extends `Model` class. Further, Model class already has the required annotation. Seems like this PR will not help in our scenario! My bad!

**Pull request closed without merging**